src/caf.js

```diff
@@ -28,7 +28,22 @@ module.exports.signalAll = signalAll;
 
 
 function CAF(generatorFn) {
-	return function instance(tokenOrSignal,...args){
+	// allow users to introspect to understand if the function they're dealing
+	// with was generated by CAF
+	Object.defineProperty(
+		instance,
+		"isCAF",
+		{ value: true },
+	);
+	Object.defineProperty(
+		instance,
+		"generatorName",
```

Comment:
Instead of `generatorName`, what would you think about `origName`? I'm a little concerned that "generator" is a bit too *leaky* of a description.